### PR TITLE
tests: net: convert normal test to use ztest

### PR DESCRIPTION
**Pull request by punitvara**
This patch convert net test cases to use ztest framework APIs
which makes output unified to other test cases.

Signed-off-by: Punit Vara <punit.vara@intel.com>

### CODE REVIEW

**Review by jukkar — CHANGES_REQUESTED** [review]
Was there some reasoning why you split the stuff to main.c and then <component>.c? To me, this does not give much benefit.

**Comment by punitvara** [comment]
@jukkar Done. Any problems to get it merge ?

File: tests/net/dhcpv4/src/dhcpv4.c [hunk 1]
```diff
@@ -527,7 +528,7 @@ static void receiver_cb(struct net_mgmt_event_callback *cb,
 	test_result(true);
 }
 
-void main(void)
+void main_dhcp(void)
```
Comment (nashif) [inline tests/net/dhcpv4/src/dhcpv4.c]: rename this, call this test_dhcp or whatever you are testing.

File: tests/net/dhcpv4/src/Makefile [hunk 2]
```diff
@@ -1,3 +1,4 @@
-obj-y = main.o
+obj-y = dhcpv4.o
```
Comment (nashif) [inline tests/net/dhcpv4/src/Makefile]: keep it main.o

File: tests/net/neighbor/src/Makefile [hunk 3]
```diff
@@ -1,3 +1,4 @@
-obj-y = main.o
+obj-y = neighbor.o
```
Comment (nashif) [inline tests/net/neighbor/src/Makefile]: keep it main.o

File: tests/net/neighbor/src/neighbor.c [hunk 4]
```diff
@@ -359,13 +358,17 @@ static bool run_tests(void)
 }
 
 
-void main(void)
+void main_neighbor(void)
```
Comment (nashif) [inline tests/net/neighbor/src/neighbor.c]: rename this, do not call this main_<something>, use the type of test you are doing as a name.

**Review by nashif — CHANGES_REQUESTED** [review]

File: tests/net/neighbor/src/neighbor.c (hunk 4 above)
Comment (punitvara) [inline tests/net/neighbor/src/neighbor.c]: Got the idea. Thanks Anas

File: tests/net/neighbor/src/Makefile (hunk 3 above)
Comment (punitvara) [inline tests/net/neighbor/src/Makefile]: Done.

File: tests/net/dhcpv4/src/Makefile (hunk 2 above)
Comment (punitvara) [inline tests/net/dhcpv4/src/Makefile]: Done.

File: tests/net/dhcpv4/src/dhcpv4.c (hunk 1 above)
Comment (punitvara) [inline tests/net/dhcpv4/src/dhcpv4.c]: Done.

**Review by jukkar — APPROVED** [review]
LGTM

**Review by nashif — APPROVED** [review]

**Pull request merged**